crates/node/src/node/grpc/mod.rs

```diff
@@ -165,6 +152,24 @@ impl miden_private_transport_proto::miden_private_transport::miden_private_trans
         Ok(tonic::Response::new(FetchNotesResponse { notes: proto_notes }))
     }
 
+    type StreamNotesStream = Sub;
+    #[tracing::instrument(skip(self), fields(operation = "grpc.stream_notes.request"))]
+    async fn stream_notes(
+        &self,
+        request: tonic::Request<StreamNotesRequest>,
+    ) -> Result<tonic::Response<Self::StreamNotesStream>, tonic::Status> {
+        let request_data = request.into_inner();
+        let tag = request_data.tag;
+        let id = rand::rng().random();
+        let (sub_tx, sub_rx) = mpsc::unbounded_channel();
```

Comment:
It is better to have some limit here and explicitly fail, otherwise it may fail with OOM killer